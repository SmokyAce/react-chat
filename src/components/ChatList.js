import React from 'react';
import classnames from 'classnames';
import { withStyles } from 'material-ui/styles';
import { Link } from 'react-router-dom';
import List, { ListItem, ListItemText } from 'material-ui/List';
import Typography from 'material-ui/Typography';
import Avatar from './Avatar';
import moment from 'moment';

const styles = theme => ({
  chatsList: {
    height: 'calc(100% - 56px)',
    overflowY: 'scroll'
  },
  active: {
    backgroundColor: '#eeeeee'
  },
  itemWrapper: {
    paddingLeft: theme.spacing.unit * 2
  }
});

const ChatList = ({ classes, chats, setActiveChat, activeId }) => {
  return (
    <List className={classes.chatsList}>
      {chats.map(chat => (
        <ListItem
          className={classnames(activeId === chat._id && classes.active)}
          key={chat._id}
          component={Link}
          to={`/chat/${chat._id}`}
          button
          onClick={() => setActiveChat(chat._id)}
        >
          <Avatar colorFrom={chat._id}>{chat.title}</Avatar>
          <div className={classes.itemWrapper}>
            <ListItemText primary={chat.title} />
            <Typography variant='caption'>
              {moment(chat.createdAt).fromNow()}
            </Typography>
          </div>
        </ListItem>
      ))}
    </List>
  );
};

export default withStyles(styles)(ChatList);
