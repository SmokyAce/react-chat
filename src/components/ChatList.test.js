/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from './ChatList';

jest.mock('./ChatListItem', () => ({ chatId }) => `ChatListItem-${chatId}`);
// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  chats: [
    {
      _id: '1',
      title: 'test',
      createdAt: '2018-03-08T13:57:58.706Z',
    },
    {
      _id: '2',
      title: 'test2',
      createdAt: '2018-03-08T13:57:58.706Z',
    },
  ],
  activeChat: {
    _id: '1',
  },
  disabled: false,
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<ChatList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without chats', () => {
    const tree = renderer.create(<ChatList {...mockProps} chats={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
