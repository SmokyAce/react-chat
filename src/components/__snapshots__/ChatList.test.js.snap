// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatsList-7"
>
  ChatListItem-1
  ChatListItem-2
</ul>
`;

exports[`<ChatList /> renders without chats 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatList-chatsList-7"
>
  <h6
    className="MuiTypography-root-13 MuiTypography-subtitle1-31 ChatList-noChats-8"
  >
    There is no chats yet...
  </h6>
</ul>
`;
